### PR TITLE
Add U2F transport-hint extension to attestation cert

### PR DESCRIPTION
**Pull request by btoews**
There's [an extension](https://fidoalliance.org/specs/fido-u2f-v1.1-id-20160915/fido-u2f-authenticator-transports-extension-v1.1-id-20160915.html) to the U2F spec that allows authenticators to hint to RPs what transports they support by adding an X509 extension to the attestation cert. This PR hints Soft U2F is "USB internal".

Also, @gregose was giving me a hard time about the `CN=mastahyeti` in the cert, so I'm switching that to `CN=Soft U2F/O=GitHub Inc./OU=Security`.

### CODE REVIEW

File: SelfSignedCertificate/SelfSignedCertificate.m [hunk 1]
```diff
@@ -23,27 +23,33 @@
...
+  "\xac\xf8\xe0\xc6\xb2\xb3\xe3\x52\x18\x3d\x80\x39\x1f\x6b\xd2\x79"
+  "\xd2\x6a\x4c\x83\x64\x74\xe6\xc2\xda\x23\x93\xff\xac\x1d\x50\x34"
+  "\x6c\x5c\x23\x90\x65\x57\x93\x3e\xcb\x93\xff\x6e\xde\xd1\xa3\x17"
+  "\x30\x15\x30\x13\x06\x0b\x2b\x06\x01\x04\x01\x82\xe5\x1c\x02\x01"
+  "\x01\x04\x04\x03\x02\x03\x08\x30\x0a\x06\x08\x2a\x86\x48\xce\x3d"
+  "\x04\x03\x02\x03\x48\x00\x30\x45\x02\x21\x00\xf6\xc8\xfd\x40\xc2"
+  "\x32\xb4\xd2\xa6\x15\x24\x10\x44\x5f\xb0\x78\x6c\x63\x69\x74\xd4"
+  "\xfd\xf0\x53\xc5\x46\xbe\xc9\x9a\xc7\x02\x67\x02\x20\x40\xb5\xf1"
+  "\xa4\x5f\xc7\xfc\x06\xd4\x88\x9d\xde\xfb\x8c\xdc\x23\xba\xb0\x42"
+  "\xd9\xf0\x83\x3d\xfd\x06\x29\x7f\x07\x9d\x81\xf1\x8a";
```
Comment (gregose) [inline SelfSignedCertificate/SelfSignedCertificate.m]: Does it make sense to dump the text output of this cert in a comment to make it clearer as to what's going on?

File: script/generate_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# X509 Extension OID for specifying supported transports.
+U2F_TRANSPORT_EXTENSION_OID = "1.3.6.1.4.1.45724.2.1.1"
+
+# From "FIDO U2F Authenticator Transports Extension" spec.
+# BIT STRING values for U2F_TRANSPORT_EXTENSION_OID.
+U2F_TRANSPORT_BLUETOOTH_RADIO            = 0b10000000
+U2F_TRANSPORT_BLUETOOTH_LOW_ENERGY_RADIO = 0b01000000
+U2F_TRANSPORT_USB                        = 0b00100000
+U2F_TRANSPORT_NFC                        = 0b00010000
+U2F_TRANSPORT_USB_INTERNAL               = 0b00001000
```
Comment (gregose) [inline script/generate_certificate.rb:26]: I didn't "USB Internal" referenced in the mentioned spec: https://fidoalliance.org/specs/fido-u2f-v1.1-id-20160915/fido-u2f-authenticator-transports-extension-v1.1-id-20160915.html#h4_fido-u2f-certificate-transports-extension. Is there some additional spec for this value?

**Review by gregose — APPROVED** [review]
lgtm, though I'll always still trust `CN=mastahyeti` :smile:

File: script/generate_certificate.rb (hunk 2 above)
Comment (btoews) [reply script/generate_certificate.rb:26]: It's a new value. One of the Google folks emailed me about it, asking for this change.

Comment (yackermann) [reply script/generate_certificate.rb:26]: @gregose The values coming from this section in specs:

```
FIDOU2FTransports ::= BIT STRING {
  bluetoothRadio(0), -- Bluetooth Classic
  bluetoothLowEnergyRadio(1),
  uSB(2),
  nFC(3)
}
```

The last value :
```
+U2F_TRANSPORT_USB_INTERNAL               = 0b00001000
```

Needs to be removed from the code, as it's not in the specs... yet

Comment (apowers313) [reply script/generate_certificate.rb:26]: I'll get the new specs posted shortly so that you can see @herrjemand 's comment in context.

Comment (apowers313) [reply script/generate_certificate.rb:26]: Latest specs here:

https://fidoalliance.org/specs/fido-u2f-v1.2-ps-20170411/

Includes the new [Authenticator Transports Extension](https://fidoalliance.org/specs/fido-u2f-v1.2-ps-20170411/fido-u2f-authenticator-transports-extension-v1.2-ps-20170411.html) spec.

**Pull request merged**